sway-lib-std/src/auth.sw

```diff
@@ -192,6 +200,60 @@ pub fn caller_address() -> Result<Address, AuthError> {
     }
 }
 
+/// Get the owners of the inputs (of type `Input::Coin` or `Input::Message`) to a
+/// `TransactionScript`.
+///
+/// # Additional Information
+///
+/// The list is not deduplicated, so there may be repeated addresses in the returned vector.
+///
+/// # Returns
+///
+/// * [Vec<Address>] - The addresses of the owners of the inputs.
+///
+/// # Examples
+///
+/// ```sway
+/// use std::auth::caller_addresses;
+///
+/// fn foo(some_address: Address) {
+///     let addresses = caller_addresses();
+///
+///     assert(addresses.get(0).unwrap() == some_address);
+/// }
+/// ```
+pub fn caller_addresses() -> Vec<Address> {
+    let inputs = input_count().as_u64();
+    let mut addresses = Vec::new();
+    let mut iter = 0;
+
+    while iter < inputs {
+        let type_of_input = input_type(iter);
+        match type_of_input {
```

Comment:
Why the outer match? Shouldn't it work exactly the same if we have only the inner match :thinking:
```
let input_owner = match input_type {
   ...
}

if let Some(address) = input_owner {
    addresses.push(address);
}

iter +=1;
```